variables.tf

```diff
@@ -14,137 +14,182 @@
   description = "The password for the default user. If not set, a random one will be generated and available in the generated_user_password output variable."
+  type        = string
   default     = ""
 }
 
-variable activation_policy {
+variable "activation_policy" {
   description = "This specifies when the instance should be active. Can be either `ALWAYS`, `NEVER` or `ON_DEMAND`."
+  type        = string
   default     = "ALWAYS"
 }
 
-variable authorized_gae_applications {
+variable "authorized_gae_applications" {
   description = "A list of Google App Engine (GAE) project names that are allowed to access this instance."
-  type        = "list"
+  type        = list(string)
   default     = []
 }
 
-variable disk_autoresize {
+variable "disk_autoresize" {
   description = "Second Generation only. Configuration to increase storage size automatically."
+  type        = bool
   default     = true
 }
 
-variable disk_size {
+variable "disk_size" {
   description = "Second generation only. The size of data disk, in GB. Size of a running instance cannot be reduced but can be increased."
+  type        = number
   default     = 10
 }
 
-variable disk_type {
+variable "disk_type" {
   description = "Second generation only. The type of data disk: `PD_SSD` or `PD_HDD`."
+  type        = string
   default     = "PD_SSD"
 }
 
-variable pricing_plan {
+variable "pricing_plan" {
   description = "First generation only. Pricing plan for this instance, can be one of `PER_USE` or `PACKAGE`."
+  type        = string
   default     = "PER_USE"
 }
 
-variable replication_type {
+variable "replication_type" {
   description = "Replication type for this instance, can be one of `ASYNCHRONOUS` or `SYNCHRONOUS`."
+  type        = string
   default     = "SYNCHRONOUS"
 }
 
-variable database_flags {
+variable "database_flags" {
   description = "List of Cloud SQL flags that are applied to the database server"
+  type        = list(map(string))
   default     = []
 }
 
-variable backup_configuration {
+variable "backup_configuration" {
   description = "The backup_configuration settings subblock for the database setings"
-  type        = "map"
-  default     = {}
-}
-
-variable ip_configuration {
+  type = object({
+    binary_log_enabled = bool
+    enabled            = bool
+    start_time         = string
+  })
+  default = {
+    binary_log_enabled = false
+    enabled            = false
+    start_time         = ""
+  }
+}
+
+variable "ip_configuration" {
   description = "The ip_configuration settings subblock"
-  type        = "list"
-  default     = [{}]
-}
-
-variable location_preference {
+  type = object({
+    authorized_networks = list(map(string))
+    ipv4_enabled        = bool
+    private_network     = string
+    require_ssl         = bool
+  })
+  default = {
+    authorized_networks = []
+    ipv4_enabled        = "true"
+    private_network     = "true"
+    require_ssl         = "false"
+  }
+}
+
+variable "location_preference" {
```

Comment:
I do not agree with you because of `follow_gae_application` and `zone` have the same type, please take a look:

						"location_preference": {
							Type:     schema.TypeList,
							Optional: true,
							MaxItems: 1,
							Computed: true,
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"follow_gae_application": {
										Type:     schema.TypeString,
										Optional: true,
									},
									"zone": {
										Type:     schema.TypeString,
										Optional: true,
									},
								},
							},
						},